backend/app/adapter/db/search.go

```diff
@@ -0,0 +1,64 @@
+package db
+
+import (
+	"database/sql"
+	"fmt"
+	"short/app/adapter/db/table"
+	"short/app/entity"
+	"short/app/usecase/search"
+)
+
+var _ search.Search = (*Search)(nil)
+
+// Search struct
+type Search struct {
+	db *sql.DB
+}
+
+// GetURLByType - to retrieve the url associated with userEmail based on type specified
```

Comment:
Please use complete sentences for comments as mentioned here: https://golang.org/doc/effective_go.html#commentary